Add common device identifier fields to aid unique identification of Linux, Mac and Windows computers

antchan2 opened the pull request:
#### Related Issue: 
https://github.com/ocsf/ocsf-schema/issues/1248

#### Description of changes:

This change adds two optional identifier fields to the [Device object](https://schema.ocsf.io/1.3.0/objects/device) to aid combining telemetry for the same device from multiple sources :

1. `os_machine_uuid`:

![image](https://github.com/user-attachments/assets/7f847366-2620-4e50-98f8-04a42ee31bb8)

2. `device_hw_info.uuid`

![image](https://github.com/user-attachments/assets/233f0d35-f385-480b-92ce-f98faae1a4a5)

pagbabian-splunk commented:
> Could we remove the win and nix specific attributes out of the extension and just make them part of core `os`? We already have OS-specific attributes there after all

The way I would suggest we do this: given that both machine identifiers are of type `uuid_t` I would just add one attribute, to device, `machine_uuid` (or `machine_uid` but NOT `machine_id` which we reserve for enums). In the description we can have both examples of where they are found, i.e. one for linux and one for Windows. Alternatively, these may be OS instance IDs, in which case we might consider that the `machine_uuid` is added to `OS`. Of course, the line is blurred in `device`.

antchan2 commented:
Thanks for the feedback, @jonrau-at-queryai, @pagbabian-splunk. 

In commit [778e083](https://github.com/ocsf/ocsf-schema/pull/1268/commits/778e0832472f94d09fe642c2e8fb57e181c2db13), I consolidated the separate Windows and Linux Machine ID fields into a single `device.os_machine_uuid` field. This avoids extensions. I believe this field is a better fit in the `device` object than in `os because while the value is assigned by the operating system, it describes the device and not the OS. I added an `os_` prefix to the field name to more clearly communicate it is for the OS-assigned value, so one should also consult the `os` field for proper interpretation.

Let me know if this fully addresses your concerns.
